include/alp/omp/storage.hpp

```diff
@@ -403,7 +424,7 @@ namespace alp {
 					const typename Distribution::GlobalCoord global( imf_r.map( i ), imf_c.map( j ) );
 					const typename Distribution::LocalCoord local = distribution.mapGlobalToLocal( global );
 
-					const size_t thread = local.tr * distribution.getThreadGridDims().second + local.tc;
+					const size_t thread = local.tr * distribution.getThreadGridDims().Tc + local.tc;
```

Comment:
within (25D_?)Distribution consider: 
- having ThreadGrid object constructed 
- exchanging ThreadCoords for specific threads in the grid